File: src/blossom/lang/Node.java
package blossom.lang;

public class Node {

    public final int id;
    private Label label;
    
    public Node(int id, Label label) {
        this.id = id;
        this.label = label;
    }
    
    public Node(int id) {
        this(id, null);
    }
    
    public Label label() {
        return label;
    }
    
}

File: ruby_interpreter/objects/node.rb
class Node

    attr_reader :id
    attr_accessor :label

    def initialize(id, label)
        @id = id
        @label = label
    end

    def to_s
        str = @id.to_s
        str += " " + @label.to_s if !@label.nil?
        return str
    end

    def clone
        return Node.new(@id, @label&.clone)
    end

end